traceon/field.py

```diff
@@ -707,5 +917,120 @@ def get_low_level_trace_function(self):
+
+    def potential_at_point(self, point):
+        elec, mag = self.is_electrostatic(), self.is_magnetostatic()
+
+        if elec and not mag:
+            return self.electrostatic_potential_at_point(point)
+        elif not elec and mag:
+            return self.magnetostatic_potential_at_point(point)
+        
+        raise RuntimeError("Cannot use potential_at_point when both electric and magnetic fields are present, " \
+            "use electrostatic_potential_at_point or magnetostatic_potential_at_point")
+        
+    def electrostatic_field_at_point(self, point):
+        return np.sum([f.electrostatic_field_at_point(point) for f in self.fields], axis=0)
+
+    def magnetostatic_field_at_point(self, point):
+        return np.sum([f.magnetostatic_field_at_point(point) for f in self.fields], axis=0)
+    
+    def current_field_at_point(self, point):
+        return np.sum([f.current_field_at_point(point) for f in self.fields], axis=0)
+
+    def electrostatic_potential_at_point(self, point):
+        return sum([f.electrostatic_potential_at_point(point) for f in self.fields])
+
+    def magnetostatic_potential_at_point(self, point):
+        return sum([f.magnetostatic_potential_at_point(point) for f in self.fields])
+    
+    def electrostatic_field_at_local_point(self, point):
+        return self.electrostatic_field_at_point(point)
+
+    def magnetostatic_field_at_local_point(self, point):
+        return self.magnetostatic_field_at_point(point)
+    
+    def current_field_at_local_point(self, point):
+        return self.current_field_at_point(point)
+
+    def electrostatic_potential_at_local_point(self, point):
+        return self.electrostatic_potential_at_point(point)
+
+    def magnetostatic_potential_at_local_point(self, point):
+        return self.magnetostatic_potential_at_point(point)
+    
+    def is_electrostatic(self):
+        return any(f.is_electrostatic() for f in self.fields)
+    
+    def is_magnetostatic(self):
+        return any(f.is_magnetostatic() for f in self.fields)
+
+    def get_tracer(self, bounds):
+        return T.Tracer(self, bounds)
+
+    def __add__(self, other):
+        if isinstance(other, (Field, FieldSuperposition)):
+            other_fields = [other] if isinstance(other, Field) else other.fields
+            fields_copy = self.fields.copy()
+            for of in other_fields:
+                for f in fields_copy:
+                    if f.matches_geometry(of):
+                        f = f + of
+                        break
+                else:
+                    fields_copy.append(of)
+
+            return FieldSuperposition(fields_copy)
+
+        return NotImplemented
+    
+    def __iadd__(self, other):
+        if isinstance(other, (Field, FieldSuperposition)):
+            other_fields = [other] if isinstance(other, Field) else other.fields
+            for of in other_fields:
+                for f in self.fields:
+                    if f.matches_geometry(of):
```

Comment:
Here you assume that `Field` contains a `matches_geometry` method. However, you didn't define any in the top level `Field` class